legacy/routes/lros.js

```diff
@@ -84,6 +86,95 @@ var lros = function (coverage) {
       .end('{ "properties": { "provisioningState": "Succeeded"}, "id": "100", "name": "foo" }');
   });
 
+  router.get("/patch/201/retry/onlyAsyncHeader", function (req, res, next) {
+    res
+      .status(200)
+      .type("json")
+      .end(
+        '{ "properties": { "provisioningState": "Succeeded"}, "id": "/lro/patch/201/retry/onlyAsyncHeader", "name": "foo" }',
+      );
+  });
+
+  router.patch("/patch/201/retry/onlyAsyncHeader", function (req, res, next) {
+    var headers = {
+      "Azure-AsyncOperation": getRequestBaseUrl(req) + "/lro/patch/201/retry/onlyAsyncHeader/operationStatuses/201",
+    };
+    res
+      .set(headers)
+      .status(201)
+      .type("json")
+      .end(
+        '{ "properties": { "provisioningState": "Accepted"}, "id": "/lro/patch/201/retry/onlyAsyncHeader", "name": "foo" }',
+      );
+  });
+
+  router.get("/patch/201/retry/onlyAsyncHeader/operationStatuses/201", function (req, res, next) {
+    coverage["LROPatch201WithAsyncHeader"]++;
+    var status = "Accepted";
+    if (coverage["LROPatch201WithAsyncHeader"] === 2) {
+      status = "Updating";
+    } else if (coverage["LROPatch201WithAsyncHeader"] === 3) {
```

Comment:
If I run the test twice without restarting the testserver (some pople may do "npm start" and run their tests several times), I will run into an infinite loop here, since I will do a ++ to make it 4, and the LRO will wait forever.

It's why usually we don't use the coverage table, since it's not reseted between runs where the test server is up and running.

Most testserver tests uses a cookie system to track status. If you want to keep it simple, do a 2 states alternative status (one call Accepted, one call Succeeded, one call Accepted, etc.). This will be fine in most scenario.

Reply:
fixed, thanks.